In [1]:
import pandas as pd

print("FIFA Data Analysis")
print("Pandas version:", pd.__version__)

FIFA Data Analysis
Pandas version: 3.0.5


# ⚽ FIFA Data Analysis

## Exploratory Data Analysis with Python and Pandas

This project explores a FIFA players dataset using Python and Pandas, focusing on data cleaning, transformation, exploratory data analysis (EDA), and data visualization.

### Objectives

- Explore and understand the dataset
- Identify and handle missing values
- Remove duplicate records
- Analyze players by nationality
- Analyze player age distribution
- Create player categories based on age
- Generate insights through data analysis and visualization

### Technologies

- Python
- Pandas
- Jupyter Notebook
- Matplotlib

## 1. Importing Libraries

In [ ]:

import pandas as pd

## 2. Loading the Dataset

In this section, the FIFA dataset is loaded using Pandas for further exploration and analysis.


In [2]:
df = pd.read_csv("../data/fifa.csv")

df.head()

,ID,NOME,IDADE,NACIONALIDADE,PONTUACAO_GERAL,POTENCIAL,TIME,VALOR,REMUNERACAO,PERNA_DE_PREFERÊNCIA,...,EMPRESTADO_DE_,VALIDADE_DO_CONTRATO,ALTURA,PESO,CONTROLE_DE_BOLA,ARRANCADA,AGILIDADE,EQUILÍBRIO,ENERGIA,FORCA
0,158023,L. Messi,31,Argentina,94,94,FC Barcelona,€110.5M,€565K,Esquerda,...,NaN,2021,5'7,159lbs,96.0,91.0,91.0,95.0,72.0,59.0
1,20801,Cristiano Ronaldo,33,Portugal,94,94,Juventus,€77M,€405K,Direita,...,NaN,2022,6'2,183lbs,94.0,89.0,87.0,70.0,88.0,79.0
2,190871,Neymar Jr,26,Brazil,92,93,Paris Saint-Germain,€118.5M,€290K,Direita,...,NaN,2022,5'9,150lbs,95.0,94.0,96.0,84.0,81.0,49.0
3,193080,De Gea,27,Spain,91,93,Manchester United,€72M,€260K,Direita,...,NaN,2020,6'4,168lbs,42.0,57.0,60.0,43.0,43.0,64.0
4,192985,K. De Bruyne,27,Belgium,91,92,Manchester City,€102M,€355K,Direita,...,NaN,2023,5'11,154lbs,91.0,78.0,79.0,77.0,90.0,75.0


## 3. Understanding the Dataset

Before cleaning and analyzing the data, we first explore the dataset structure, dimensions, column names, data types, and descriptive statistics.

In [3]:
# Dataset dimensions
df.shape

(18207, 24)

In [4]:
# Column names
df.columns

Index(['ID', 'NOME', 'IDADE', 'NACIONALIDADE', 'PONTUACAO_GERAL', 'POTENCIAL',
       'TIME', 'VALOR', 'REMUNERACAO', 'PERNA_DE_PREFERÊNCIA',
       'REPUTACAO_INTERNACIONAL', 'POSICAO', 'CAMISA', 'CONTRATACAO',
       'EMPRESTADO_DE_', 'VALIDADE_DO_CONTRATO', 'ALTURA', 'PESO',
       'CONTROLE_DE_BOLA', 'ARRANCADA', 'AGILIDADE', 'EQUILÍBRIO', 'ENERGIA',
       'FORCA'],
      dtype='str')

In [5]:
# Dataset information
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 18207 entries, 0 to 18206
Data columns (total 24 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   ID                       18207 non-null  int64  
 1   NOME                     18207 non-null  str    
 2   IDADE                    18207 non-null  int64  
 3   NACIONALIDADE            18207 non-null  str    
 4   PONTUACAO_GERAL          18207 non-null  int64  
 5   POTENCIAL                18207 non-null  int64  
 6   TIME                     17966 non-null  str    
 7   VALOR                    18207 non-null  str    
 8   REMUNERACAO              18207 non-null  str    
 9   PERNA_DE_PREFERÊNCIA     18207 non-null  str    
 10  REPUTACAO_INTERNACIONAL  18159 non-null  float64
 11  POSICAO                  18207 non-null  str    
 12  CAMISA                   18147 non-null  float64
 13  CONTRATACAO              16654 non-null  str    
 14  EMPRESTADO_DE_           1264 non

In [6]:
# Descriptive statistics
df.describe()

,ID,IDADE,PONTUACAO_GERAL,POTENCIAL,REPUTACAO_INTERNACIONAL,CAMISA,CONTROLE_DE_BOLA,ARRANCADA,AGILIDADE,EQUILÍBRIO,ENERGIA,FORCA
count,18207.000000,18207.000000,18207.000000,18207.000000,18159.000000,18147.000000,18159.000000,18159.000000,18159.000000,18159.000000,18159.000000,18159.000000
mean,214298.338606,25.122206,66.238699,71.307299,1.113222,19.546096,58.369459,64.614076,63.503607,63.966573,63.219946,65.311967
std,29965.244204,4.669943,6.908930,6.136496,0.394031,15.947765,16.686595,14.927780,14.766049,14.136166,15.894741,12.557000
min,16.000000,16.000000,46.000000,48.000000,1.000000,1.000000,5.000000,12.000000,14.000000,16.000000,12.000000,17.000000
25%,200315.500000,21.000000,62.000000,67.000000,1.000000,8.000000,54.000000,57.000000,55.000000,56.000000,56.000000,58.000000
50%,221759.000000,25.000000,66.000000,71.000000,1.000000,17.000000,63.000000,67.000000,66.000000,66.000000,66.000000,67.000000
75%,236529.500000,28.000000,71.000000,75.000000,1.000000,26.000000,69.000000,75.000000,74.000000,74.000000,74.000000,74.000000
max,246620.000000,45.000000,94.000000,95.000000,5.000000,99.000000,96.000000,97.000000,96.000000,96.000000,96.000000,97.000000


### Initial Observations

In [7]:
df.shape
df.info()
df.describe()

<class 'pandas.DataFrame'>
RangeIndex: 18207 entries, 0 to 18206
Data columns (total 24 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   ID                       18207 non-null  int64  
 1   NOME                     18207 non-null  str    
 2   IDADE                    18207 non-null  int64  
 3   NACIONALIDADE            18207 non-null  str    
 4   PONTUACAO_GERAL          18207 non-null  int64  
 5   POTENCIAL                18207 non-null  int64  
 6   TIME                     17966 non-null  str    
 7   VALOR                    18207 non-null  str    
 8   REMUNERACAO              18207 non-null  str    
 9   PERNA_DE_PREFERÊNCIA     18207 non-null  str    
 10  REPUTACAO_INTERNACIONAL  18159 non-null  float64
 11  POSICAO                  18207 non-null  str    
 12  CAMISA                   18147 non-null  float64
 13  CONTRATACAO              16654 non-null  str    
 14  EMPRESTADO_DE_           1264 non

,ID,IDADE,PONTUACAO_GERAL,POTENCIAL,REPUTACAO_INTERNACIONAL,CAMISA,CONTROLE_DE_BOLA,ARRANCADA,AGILIDADE,EQUILÍBRIO,ENERGIA,FORCA
count,18207.000000,18207.000000,18207.000000,18207.000000,18159.000000,18147.000000,18159.000000,18159.000000,18159.000000,18159.000000,18159.000000,18159.000000
mean,214298.338606,25.122206,66.238699,71.307299,1.113222,19.546096,58.369459,64.614076,63.503607,63.966573,63.219946,65.311967
std,29965.244204,4.669943,6.908930,6.136496,0.394031,15.947765,16.686595,14.927780,14.766049,14.136166,15.894741,12.557000
min,16.000000,16.000000,46.000000,48.000000,1.000000,1.000000,5.000000,12.000000,14.000000,16.000000,12.000000,17.000000
25%,200315.500000,21.000000,62.000000,67.000000,1.000000,8.000000,54.000000,57.000000,55.000000,56.000000,56.000000,58.000000
50%,221759.000000,25.000000,66.000000,71.000000,1.000000,17.000000,63.000000,67.000000,66.000000,66.000000,66.000000,67.000000
75%,236529.500000,28.000000,71.000000,75.000000,1.000000,26.000000,69.000000,75.000000,74.000000,74.000000,74.000000,74.000000
max,246620.000000,45.000000,94.000000,95.000000,5.000000,99.000000,96.000000,97.000000,96.000000,96.000000,96.000000,97.000000


### Initial Observations

- The dataset contains 18,207 players and 24 columns.
- Player ages range from 16 to 45 years, with an average age of approximately 25 years.
- The average overall rating is approximately 66.2, while the average potential rating is approximately 71.3.
- Some columns contain missing values and will require further investigation and data cleaning.
- The dataset contains both numerical and categorical variables, which will allow different types of exploratory analysis.